.github/workflows/release.yml

```diff
@@ -31,15 +29,52 @@ jobs:
         run: |
           # Remove dist, build, and pylint.egg-info
           # when building locally for testing!
-          python -m pip install twine build
+          python -m pip install build
       - name: Build distributions
         run: |
           python -m build
+      - name: Upload release assets
+        uses: actions/upload-artifact@v4.6.1
+        with:
+          name: release-assets
+          path: dist/
+
+  release-pypi:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    environment:
+      name: PyPI
+      url: https://pypi.org/project/pylint/
+    permissions:
+      id-token: write
+    steps:
+      - name: Download release assets
+        uses: actions/download-artifact@v4.1.9
+        with:
+          name: release-assets
+          path: dist/
       - name: Upload to PyPI
         if: github.event_name == 'release' && startsWith(github.ref, 'refs/tags')
-        env:
-          TWINE_REPOSITORY: pypi
-          TWINE_USERNAME: __token__
-          TWINE_PASSWORD: ${{ secrets.PYPI_API_TOKEN }}
-        run: |
-          twine upload --verbose dist/*
+        uses: pypa/gh-action-pypi-publish@release/v1
+
+  release-github:
+    name: Upload assets to Github release
+    runs-on: ubuntu-latest
+    needs: ["build"]
+    permissions:
+      contents: write
+      id-token: write
+    steps:
+      - name: Download release assets
+        uses: actions/download-artifact@v4.1.9
+        with:
+          name: release-assets
+          path: dist/
+      - name: Sign the dists with Sigstore and upload assets to Github release
+        if: github.event_name == 'release' && startsWith(github.ref, 'refs/tags')
```

Comment:
Why not skip the entire job?

Reply:
The job shouldn't even start in the first place.
- The workflow is only triggered on `release -> published`
- The upload jobs (to Github and PyPI) need `Build` which is only run for `github.event_name == 'release'`

I usually prefer to be a bit more explicit with these checks when it comes to releases, even if not really necessary.

Reply:
Yeah, I had a feeling this was overly verbose..